### PR TITLE
Add component for SPECTRUM Tango data

### PR DESCRIPTION
**Pull request by GregViguier**
Hi Michal,
I made a component for SPECTRUM. It's a hacked version of your trend component.
I'm a total beginner in JS&HTML so feel free to enhance my code.

Cheers,

Greg

### CODE REVIEW

**Comment by mliszcz** [comment]
Thanks Greg!

Awesome contribution. Your code looks OK to me. I will try to run it tomorrow (I have some issues after upgrading to mTango rc3, need to sort out that first), then I will merge this PR.

Consider adding some notes about new widgets to the README, so anyone interested will be able to find them.

Best regards,
Michal

**Comment by mliszcz** [comment]
In the meantime, if you are interested, here is my thesis (from Dec 2016):
https://github.com/mliszcz/msc-thesis/raw/master/thesis/pdf/msc-thesis-michal-liszcz-tangojs.pdf

Appendices contain some technical details and can serve as a documentation.

**Comment by GregViguier** [comment]
Thanks for the link. I read the complete document. It's clearer now in my mind.
The Image component should not be a part of this pull request. It's a work in progress from my apprentice.

File: dist/components/tangojs-image.html [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+      this.root.appendChild(this.dom.grid)
+
+      //TODO instancier composant ici
+      this.viewer = this.createViewer(this.dom.grid)
+
+      this.onResizeCallback = () => {
+
+      }
+
+      this.mutationObserver = new MutationObserver(this.onResizeCallback)
```
Comment (mliszcz) [inline dist/components/tangojs-image.html:28]: Maybe you don't need a MutationObserver here. Can attributeChangedCallback do the job here?
(see: https://developer.mozilla.org/en-US/docs/Web/Web_Components/Custom_Elements)

If you just want to listen for changes in attribute values, you can also use `mixins.withReflectedAttributes` - it automagically maps HTML attributes to javascript properties of `this` (the widget) and takes care of conversions to/from DOM strings.

**Comment by mliszcz** [comment]
Hi Greg,

I've tried new `tangojs-chart` with TangoTest device and it worked fine. I've merged the PR, updated the README with information about new widgets and released new version to npm (@0.4.6). Once again thanks for this contribution!

If you have any questions regarding the `image` widget, I wll be glad to help.

Cheers,
Michal

**Pull request merged**